### PR TITLE
Update Developer Guide documentation and Testing

### LINKED ISSUE #100: Update Developer Guide to include new methods from v1.2-v1.3

### PR DESCRIPTION
**Pull request by L1uY1jun**

### CODE REVIEW

File: src/main/java/seedu/address/logic/parser/FilterCommandParser.java [hunk 1]
```diff
@@ -18,16 +19,13 @@ public class FilterCommandParser implements Parser<FilterCommand> {
      */
     public FilterCommand parse(String args) throws ParseException {
         String trimmedArgs = args.toLowerCase().trim();
-        switch (trimmedArgs) {
-        case "pending":
-        case "accepted":
-        case "rejected":
+
+        if (ApplicationStatus.isValidApplicationStatus(trimmedArgs)) {
```
Comment (ShummyOwnzYou) [inline src/main/java/seedu/address/logic/parser/FilterCommandParser.java]: Perhaps consider using the if block to throw the exception instead? This way the parser deals with unexpected conditions first before moving on to the main path.

**Review by ShummyOwnzYou — COMMENTED** [review]
LGTM, apart from some minor details

File: src/main/java/seedu/address/logic/parser/FilterCommandParser.java (hunk 1 above)
Comment (L1uY1jun) [reply src/main/java/seedu/address/logic/parser/FilterCommandParser.java]: Noted, will make ammendments

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -214,28 +215,67 @@ The filter operation is facilitated by `FilterCommand`. It extends `Command` and
...
+Given below is an example usage scenario and how the remove operation is handled by TrackAScholar:
+
+1. The user enters `remove accepted`, for example, to remove all applicants with accepted scholarship application status.
+   This invokes `LogicManager#execute()`, which calls `TrackAScholarParser#parseCommand()` to separate the command word `remove` and
+   the argument `accepted`.
+
+2. `TrackAScholarParser` identifies the `remove` command and `RemoveCommandParser` will be instantiated which calls `RemoveCommandParser#parse()`.
+
+3. `RemoveCommandParser#parse()` now parses the argument and creates a new `ApplicationStatus` before finally initializing and returning a `RemoveCommand`
+   with the new `ApplicationStatus` as an argument.
```
Comment (benjytan45678) [inline docs/DeveloperGuide.md:259]: Is this an intentional new line here?

File: docs/DeveloperGuide.md [hunk 3]
```diff
@@ -214,28 +215,67 @@ The filter operation is facilitated by `FilterCommand`. It extends `Command` and
...
+
+4. `LogicManager#execute()` now calls `RemoveCommand#execute()`, which invokes `RemoveCommand#promptUserConfirmation()`. TrackAScholar now displays
+   a window asking for the user's confirmation to remove the applicants. After the use confirms, `RemoveCommand#confirmRemove()` is called which
```
Comment (benjytan45678) [inline docs/DeveloperGuide.md]: you mean user, not use?

Comment (L1uY1jun) [reply docs/DeveloperGuide.md]: Yes! Thanks for spotting

**Pull request closed without merging**